powerline/segments/i3wm.py

```diff
@@ -33,33 +42,64 @@ def workspaces(pl, segment_info, only_show=None, output=None, strip=0):
 		on that output are shown. Overrides automatic output detection by 
 		the lemonbar renderer and bindings.
 
-	:param int strip:
-		Specifies how many characters from the front of each workspace name 
-		should be stripped (e.g. to remove workspace numbers). Defaults to zero.
+	:param bool strip:
+		If set to True, workspaces prefixed with a number and colon 
+		(like ``1: www`` or ``1:www``) will have only the name shown (``www``).
+		Default: False.
 
 	Highlight groups used: ``workspace`` or ``w_visible``, ``workspace`` or ``w_focused``, ``workspace`` or ``w_urgent``.
 	'''
-	global conn
-	if not conn:
-		try:
-			import i3ipc
-		except ImportError:
-			import i3 as conn
-		else:
-			conn = i3ipc.Connection()
-
 	output = output or segment_info.get('output')
 
+	if type(strip) == int:
+		pl.warn("integer values for 'strip' are deprecated")
+
 	return [{
-		'contents': w['name'][min(len(w['name']), strip):],
+		'contents': format_name(w['name'], strip=strip),
 		'highlight_groups': calcgrp(w)
-	} for w in conn.get_workspaces()
-		if (not only_show or any(w[typ] for typ in only_show))
-		and (not output or w['output'] == output)
+	} for w in get_i3_connection().get_workspaces()
+		if ((not only_show or any(w[typ] for typ in only_show))
+		    and (not output or w['output'] == output))
 	]
 
 
 @requires_segment_info
+def workspace(pl, segment_info, workspace=None, strip=False):
+	'''Return state for the specified workspace
+
+	:param str workspace:
+		Specifies which workspace to show. If unspecified, may be set by the 
+		``list_workspaces`` lister.
+
+	:param bool strip:
+		If set to True, workspaces prefixed with a number and colon 
+		(like ``1: www`` or ``1:www``) will have only the name shown (``www``).
+		Default: False.
+
+	Highlight groups used: ``workspace`` or ``w_visible``, ``workspace`` or ``w_focused``, ``workspace`` or ``w_urgent``.
+	'''
+
+	if type(strip) == int:
+		pl.warn("integer values for 'strip' are deprecated")
```

Comment:
And this warning should be removed, and integer arguments treated as regular boolean values.